saexec/execution.go

```diff
@@ -112,7 +113,10 @@ func (e *Executor) execute(b *blocks.Block, logger logging.Logger) error {
 	}
 
 	header := types.CopyHeader(b.Header())
-	header.BaseFee = gasClock.BaseFee().ToBig()
+	// TODO(cey): this is a janky hack to pass the base fee tests in ethtests.
+	if !e.execOpts.preserveBaseFee {
+		header.BaseFee = gasClock.BaseFee().ToBig()
+	}
```

Comment:
this is a very janky workaround. Without this I could not get any tests after `London` fork to pass. Because the baseFee changes both the end state, where tests expecting a certain balance after txs and also it changes rewards accumulated in coinbase.
Alternatives are:
1- Set the gas target before execution, so that we can try to make `gasClock` equivalent to whatever in the `header.BaseFee`
2- Modify tests and expected state.